File: OOP/ACCOUNT/PYTHON/Account.py
class Account:
    def __init__(self, name, pin, balance, number):
        self.name = name
        self.pin = pin
        self.balance = balance
        self.number = number

    def get_account(self):
        return self.balance
    
    def deposit(self, amount):
        if amount > 0:
            self.balance += amount
        else:
            print("Amount Must Be Greater Than Zero.")

    def withdraw(self, amount, pin):
        k

File: OOP/ACCOUNT/JAVA/Account.java
package OOP.ACCOUNT.JAVA;

public class Account {
    private String name;
    private String pin;
    private int balance;
    private int number;

    public Account(String name, String pin, int balance, int number) {
        this.name = name;
        this.pin = pin;
        this.balance = balance;
        this.number = number;
    }

    public int getAccountNumber() {
        return this.number;
    }

    public void deposit(int amount) {
        if(amount > 0) {
            this.balance += amount;
        } else {
	    throw new IllegalArgumentException("Amount Must Be Greater Than Zero.");
	}
    }

    public void withdraw(int amount, String pin) {
        if(amount > 0 && amount <= this.balance && this.pin.equals(pin)) {
            this.balance = this.balance - amount;
        } else {
	    throw new IllegalArgumentException("Amount Must Be Greater Than Zero."); 
	}
    }

    public int checkBalance(String pin) {
        if(this.pin.equals(pin)) {
	    return this.balance;
	} else {
	    throw new IllegalArgumentException("Incorrect Pin.");
	}
    }
}
